src/outfit.ts

```diff
@@ -257,19 +257,34 @@ export class Outfit {
   }
 
   /**
-   * Adds a value to any existing bonus this item has; if it started without a bonus, sets the bonus equal to that value.
+   * Adds a value to any existing bonus this item has, using a rule assigned by the `reducer` parameter
    *
    * Only triggers on items that may be equipped to this outfit.
    * @param item The item to try to add a bonus to.
    * @param value The value to try to add.
+   * @param reducer Function that combines new and current bonus
    * @returns The total assigned bonus to that item.
    */
-  public addBonus(item: Item, value: number): number {
+  public addBonus(
+    item: Item,
+    value: number,
+    reducer = (a: number, b: number) => a + (b ?? 0)
```

Comment:
Given that `getBonus` always returns a number, shouldn't this just be `a + b`?
```suggestion
    reducer = (a: number, b: number) => a + b
```